fontforge/featurefile.c

```diff
@@ -2435,21 +2438,43 @@ return;
 return;
     }
 
-    if ( *namebuf=='/' ||
-	    ( pt = strrchr(tok->filename[tok->inc_depth],'/') )==NULL )
-	filename=copy(namebuf);
-    else {
-	*pt = '\0';
-	filename = GFileAppendFile(tok->filename[tok->inc_depth],namebuf,false);
-	*pt = '/';
+
+    pt = strrchr(tok->filename[tok->inc_depth],'/');
+    if ( pt != NULL ) *pt = '\0';
+    filename_deprecated = GFileAppendFile(tok->filename[tok->inc_depth],namebuf,false);
+    if ( pt != NULL ) *pt = '/';
+    if ( tok->in_ufo ) {
+        // we know this is a features.fea file
+        filename = copy(tok->filename[tok->inc_depth]);
```

Comment:
I know the spec page you linked to says  "Any `include()` statements must be relative to the UFO path, *not* to the features.fea file itself." but are you sure they didn't mean "Any *relative* `include()` statements ..."? That is, are you sure they're ruling out absolute includes? I'm not sure why that would be.

Reply:
I'm afraid I don't quite understand the question. Can you say what you think my patch does wrong in the case of absolute includes?

Reply:
I'm afraid I don't quite understand the question. Can you say what you think my patch does wrong in the case of absolute includes?

Reply:
The old code path had a special case for `*namebuf=='/'` where it just copied the whole filename (thereby handling absolute paths). I don't see the equivalent in the new code path.

Reply:
I don't really see any evidence absolute includes are supported in UFO. What about Windows, where it's C:, D:, etc? The old code wouldn't have worked with that. I think UFO's strong use of "any" means that even leading slashes denote relative files.

Reply:
The spec has been updated to include language about UFO resolution and it doesn't rule them out: http://adobe-type-tools.github.io/afdko/OpenTypeFeatureFileSpecification.html#3-including-files . And I'm not sure why one *would* rule them out. Most paths can be relative or absolute.

Reply:
Do I also need to support Windows notation, as `C:/WINDOWS/example.ufo/features.fea`? I _assume_ that this can happen, but that we're saying even on Windows to use <kbd>/</kbd>, as elsewhere? I'm just not sure absolute includes make any sense in a UFO file. UFO is supposed to be self-contained, movable around the system. The reason the FEA's live in the dir above is MM fonts, made of multiple UFO's under a project root. Why would you ever even have a UFO with an absolute include?

Reply:
I believe that somehow, perhaps via posix interfaces, the slash/backslash problem with pathnames is taken care of, so the windows notation issue would be related to switching drive letters. Allowing that would be virtuous but probably not mandatory.

I don't read the spec the way you do and I take the possibility of an absolute path as a potential work-around, perhaps a temporary one, that should still be available. If you want this to go in with this functionality change you should probably be asking @frank-trampe or someone else on the core team.

Reply:
It's true that on modern Windows you can mount a drive letter under any other drive letter. If we're saying that's the solution here, then I can deal with that and go ahead and update the patch to consider `/` to be absolute, even in a UFO, as before. I don't know that that's exactly to-spec, but I can see why you might think it should work, and it's probable other software supports it. The UFO spec has some ambiguities, this is one, so I can resolve it this way, that's fine, just leaving the resolution of drive letters up to the user to fix, either through mounting or what have you.